### PR TITLE
[DP-158] As a reader, I need to be able to read a chapter

### PR DESCRIPTION
**Pull request by chanmioh**
# Images

## View of a chapter page containing document content
image

## View of a subchapter page containing document content
image

## At the bottom of this chapter are its subchapters
image

# Changelog
- `query.rs`: Added a `chapter` query that retrieves an `EditedCollection`'s chapter by its collection slug and chapter slug.
- `documents_in_group.sql`: Since a DAILP document can be in a CWKW collection chapter, this queries for a `chapter_path` of a document if there is one.
- `database_sql.rs`: Documents in a collection now return an associated `chapter_path` if there is one.
- `collection.rs`: Added a resolver on `CollectionChapter` for returning its associated document content, if there is one.
- `chapter.page.tsx`: Chapter page component that renders the Wordpress prose if there is one, and renders a list of its subchapters. Subchapters render either a document or a Wordpress page.
- `chapters-context.tsx`: Added context that holds information about all the chapters in a collection along with their subchapters.

### CODE REVIEW

**Review by GracefulLemming — APPROVED** [review]
I think in general I didn't have trouble understanding how things worked. If Ellen thought the sample images were sufficient, then I say we can move this forward!

File: types/src/database_sql.rs [hunk 1]
```diff
@@ -467,13 +467,31 @@ impl Database {
...
+                id: doc.id,
+                short_name: doc.short_name,
+                title: doc.title,
+                date: doc.date,
+                order_index: doc.order_index,
+                chapter_path: if doc.chapter_path.is_none() {
+                    None
+                } else {
+                    Some(
+                        doc.chapter_path
```
Comment (loafofpiecrust) [inline types/src/database_sql.rs:486]: you can use `doc.chapter_path.map()`

File: website/src/pages/collections/collection.page.tsx [hunk 2]
```diff
@@ -36,7 +36,17 @@ const CollectionPage = () => {
...
+                <Link
+                  href={
+                    // If this document exists within a collection's chapter, then redirect to that chapter page.
+                    document.chapterPath && document.chapterPath.length > 1
+                      ? chapterRoute(
+                          document.chapterPath[0]!,
+                          document.chapterPath[document.chapterPath.length - 1]!
+                        )
+                      : documentRoute(document.slug)
+                  }
```
Comment (loafofpiecrust) [inline website/src/pages/collections/collection.page.tsx:48]: This might be better as a new function alongside `chapterRoute` and `documentRoute`

**Comment by loafofpiecrust** [comment]
The CWKW home page is at the wrong url, it should be `/cwkw` not `/collections/cwkw`

**Pull request merged**